### PR TITLE
PMM 7492: SMTP dynamic options

### PR DESCRIPTION
**Pull request by fabio-silva**
https://github.com/Percona-Lab/pmm-submodules/pull/1635

What this PR does / why we need it: adds an auth type toggle to PMM settings and shows different fields for each option.

Which issue(s) this PR fixes: https://jira.percona.com/browse/PMM-7492

# Implementation
- `Server`, `From` and `Hello` are always present;
- `Username` and `Password` are disabled for `None`;
- `Username` and `Password` are enabled for every other option;
- When the form is submitted, only the relevant fields are sent according to the selected type. `Identity` is automatically sent using a Base64 string using username+password;
- CRAM-MD5 password is sent as `secret`, keeping the API as it was;
- When settings are retrieved, the type is automatically inferred.

# Screenshots
Screenshot 2021-03-22 at 18 17 07
Screenshot 2021-03-22 at 18 17 17

### CODE REVIEW

**Review by atymchuk — APPROVED** [review]

File: public/app/percona/settings/Settings.service.ts [hunk 1]
```diff
@@ -26,12 +26,36 @@ export const SettingsService = {
     return response;
   },
   async setSettings(body: any, setLoading: LoadingCallback) {
-    let response: any;
+    let response: Settings = {
```
Comment (tiagomotasantos) [inline public/app/percona/settings/Settings.service.ts:29]: Should this be `const` now?

File: public/app/percona/settings/components/Communication/Email/Email.types.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { EmailAuthType, EmailSettings } from '../../../Settings.types';
+import { LoadingCallback } from '../../../Settings.service';
+
+export interface EmailProps {
+  settings: EmailSettings;
+  updateSettings: (body: any, callback: LoadingCallback) => void;
```
Comment (tiagomotasantos) [inline public/app/percona/settings/components/Communication/Email/Email.types.ts:6]: Should we have a type for body?

**Review by tiagomotasantos — CHANGES_REQUESTED** [review]

File: public/app/percona/settings/Settings.service.ts (hunk 1 above)
Comment (fabio-silva) [reply public/app/percona/settings/Settings.service.ts:29]: The var might be reassigned when the API responds :S

File: public/app/percona/settings/components/Communication/Email/Email.types.ts (hunk 2 above)
Comment (fabio-silva) [reply public/app/percona/settings/components/Communication/Email/Email.types.ts]: Done 👍

**Review by tiagomotasantos — DISMISSED** [review]

**Comment by fabio-silva** [comment]
(rebased after base branch force-pushed)

**Review by tiagomotasantos — APPROVED** [review]

**Pull request merged**